File: Searching/LinearSearch.js
function linearSearch(array, targetValue) {

    for (var i = 0; i < array.length; i++) {
        if (targetValue === array[i]) {
            return i;
        }
    }
    return "Not Found";
}

function main() {

    // Example
    var exampleArray = [1, 2, 3, 4, 5, 6, 7, 8, 9];
    var exampleTarget = 5
    result = linearSearch(exampleArray, exampleTarget)
    if (result === "Not Found") {
        console.log("Your value " + exampleTarget + " was " + result);
    } else {
        console.log("Your value " + exampleTarget + " was found at array index " + result)
    }
}

main();

File: Searching/LinearSearch.cpp
#include <vector>
#include <stdio.h>

int linearSearch(const std::vector<int> arr, int value)
{
  for (unsigned int i = 0; i < arr.size(); ++i)
  {
    if (arr[i] == value)
    {
      return i;
    }
  }
   return -1;
}
 
int main()
{
   std::vector<int> arr = {10, 34, 22, 4, 5};
   int x = 10;
   int result = linearSearch(arr, x);
   (result == -1)? printf("Element is not present in array")
                 : printf("Element is present at index %d", result);
   return 0;
}
